src/PoolManager.sol

```diff
@@ -135,35 +130,33 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, ERC6909Claims {
 
     /// @inheritdoc IPoolManager
     function lock(address lockTarget, bytes calldata data) external payable override returns (bytes memory result) {
-        Lockers.push(lockTarget, msg.sender);
+        // Get the lock caller because thats an EOA and is not user-controlable
+        if (Locker.isLocked()) revert AlreadyLocked();
+
+        Locker.setLocker(lockTarget);
 
         // the caller does everything in this callback, including paying what they owe via calls to settle
         result = ILockCallback(lockTarget).lockAcquired(msg.sender, data);
```

Comment:
interesting. one thing I just thought of is that there isn't really a `result` subsequent callers on the pool manager can get now. I don't think its bad, just a breaking integration change.

i guess new integrators would just ingest the return from any function they call directly

Reply:
Right but also the address who returns the result from `lockAcuired` is also the same address that called `lock` anyway (once we remove EOA calling)... so they could also just use transient storage to "return" that value to themselves. Arguably the return value isnt needed at all right?

Reply:
that is true if we remove EOA calling